src/UserSpaceInstrumentation/OrbitUserSpaceInstrumentation.cpp

```diff
@@ -104,6 +111,13 @@ void StartNewCapture() {
 }
 
 void EntryPayload(uint64_t return_address, uint64_t function_id, uint64_t stack_pointer) {
+  bool& is_in_payload = GetIsInPayload();
+  if (is_in_payload) {
+    // We already overwrote the return address with the exit trampoline. So we need to restore this.
```

Comment:
I suggest expanding this comment because it's not super clear:
"Before calling EntryPayload, we overwrote the return address with the address of the exit trampoline a second time. So we need to restore the previous return address."

Reply:
But I have a question out of curiosity: considering that there's a single exit trampoline (right? or is it per module?) would this actually be necessary? Or did I get this wrong?

Reply:
Never mind, I got it completely wrong.